Firestore/core/src/firebase/firestore/local/leveldb_index_manager.mm

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/local/leveldb_index_manager.h"
+
+#include <string>
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/local/leveldb_key.h"
+#include "Firestore/core/src/firebase/firestore/local/memory_index_manager.h"
+#include "Firestore/core/src/firebase/firestore/model/resource_path.h"
+#include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+#include "absl/strings/match.h"
+
+#import "Firestore/Source/Local/FSTLevelDB.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+using model::ResourcePath;
+
+LevelDbIndexManager::LevelDbIndexManager(FSTLevelDB *db) : db_(db) {
+}
+
+void LevelDbIndexManager::AddToCollectionParentIndex(
+    const ResourcePath &collection_path) {
+  HARD_ASSERT(collection_path.size() % 2 == 1, "Expected a collection path.");
```

Comment:
Optional: add a newline to separate the precondition check.

Reply:
done